crabs/detector/utils/detection.py

```diff
@@ -242,3 +243,74 @@ def slurm_logs_as_artifacts(logger: MLFlowLogger, slurm_job_id: str):
             logger.run_id,
             f"{log_filename}.{ext}",
         )
+
+
+def bbox_tensors_to_COCO_dict(
+    bbox_tensors: torch.Tensor, list_img_filenames: Optional[list] = None
+) -> dict:
+    """Convert list of bounding boxes as tensors to COCO-crab format.
+
+    Parameters
+    ----------
+    bbox_tensors : list[torch.Tensor]
+        List of tensors with bounding boxes for each image.
+        Each element of the list corresponds to an image, and each tensor in
+        the list contains the bounding boxes for that image. Each tensor is of
+        size (n, 4) where n is the number of bounding boxes in the image.
+        The 4 values in the second dimension are x_min, y_min, x_max, y_max.
+    list_img_filenames : list[str], optional
+        List of image filenames. If not provided, filenames are generated
+        as "frame_{i:04d}.png" where i is the 0-based index of the image in the
+        list of bounding boxes.
+
+    Returns
+    -------
+    dict
+        COCO format dictionary with bounding boxes.
+    """
+    # Create list of image filenames if not provided
+    if list_img_filenames is None:
+        list_img_filenames = [
+            f"frame_{i:04d}.png" for i in range(len(bbox_tensors))
+        ]
+
+    # Create list of dictionaries for images
+    list_images: list[dict] = []
+    for img_id, img_name in enumerate(list_img_filenames):
+        image_entry = {
+            "id": img_id + 1,  # 1-based
+            "width": 0,
+            "height": 0,
+            "file_name": img_name,
+        }
+        list_images.append(image_entry)
+
+    # Create list of dictionaries for annotations
+    list_annotations: list[dict] = []
+    for img_id, img_bboxes in enumerate(bbox_tensors):
+        # loop thru bboxes in image
+        for bbox_row in img_bboxes:
+            x_min, y_min, x_max, y_max = bbox_row.numpy().tolist()
+            # we convert the array to list to make it JSON serializable
+
+            annotation = {
+                "id": len(list_annotations) + 1,  # 1-based
+                "image_id": img_id,
```

Comment:
is the image_id is 0-based?

Reply:
thanks Nik, yes I missed this! 

It is indeed 1-based by default in VIA

🦅 👁️ !